spring-pulsar-spring-boot-autoconfigure/build.gradle

```diff
@@ -9,7 +9,9 @@ dependencies {
 	annotationProcessor 'org.springframework.boot:spring-boot-autoconfigure-processor'
 	annotationProcessor 'org.springframework.boot:spring-boot-configuration-processor'
 
-	api project (':spring-pulsar')
+	optional project (':spring-pulsar')
+	optional 'org.apache.pulsar:pulsar-client-reactive-producer-cache-caffeine'
```

Comment:
Although the cache module brings in the adapter, I would be in favor of explicitly bringing in the adapter as optional here as well.

Reply:
With comment https://github.com/spring-projects-experimental/spring-pulsar/pull/175/files#r1005889990 it will also be pulled by spring-pulsar.
If we do it for the reactive adapter, shouldn't we do it also for the regular client ?